### PR TITLE
Backport tapzero plan

### PR DESCRIPTION
**Pull request by bcomnes**
Backport https://github.com/socketsupply/tapzero/pull/16

### CODE REVIEW

**Comment by bcomnes** [comment]
cc @nichoth

**Review by jwerle — APPROVED** [review]

**Comment by chicoxyzzy** [comment]
@bcomnes please regenerate docs & type definitions with `npm run gen`

File: api/test/README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ The implementation is <250 loc, (<500 with comments) ( https://github.com/Raynos
 ## Migrating from tape
 
 ```js
-const test = require('tape')
+const tape = require('tape')
 // Tapzero exports an object with a test function property.
 const test = require('tapzero').test
 ```
```
Comment (chicoxyzzy) [inline api/test/README.md:15]: not related to this PR, but the readme looks incorrect here. `socket:test` is a built-in module.

Comment (chicoxyzzy) [reply api/test/README.md:15]: do we even need this md file?

Comment (bcomnes) [reply api/test/README.md:15]: I'm not sure the intention. Maybe we just drop it?

Comment (bcomnes) [reply api/test/README.md]: Done

**Comment by bcomnes** [comment]
Ok @chicoxyzzy I think this is good now.

**Review by jwerle — APPROVED** [review]

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**